src/test/scala/io/iohk/ethereum/blockchain/sync/SyncControllerSpec.scala

```diff
@@ -497,23 +553,24 @@ class SyncControllerSpec extends FlatSpec with Matchers {
 
     fastSyncController ! SyncController.StartSync
 
-    peer1.ignoreMsg { case u => u == Unsubscribe }
+    peer1TestProbe.expectMsg(PeerActor.SendMessage(GetBlockHeaders(Left(expectedMaxBlock + 1), Config.FastSync.blockHeadersPerRequest, 0, reverse = false)))
+    peerMessageBus.expectMsg(Subscribe(MessageClassifier(Set(BlockHeaders.code), PeerSelector.WithId(peer1.id))))
+    peerMessageBus.reply(MessageFromPeer(BlockHeaders(Seq(newBlockHeader)), peer1.id))
 
-    peer1.expectMsg(PeerActor.SendMessage(GetBlockHeaders(Left(expectedMaxBlock + 1), Config.FastSync.blockHeadersPerRequest, 0, reverse = false)))
-    peer1.expectMsg(PeerActor.Subscribe(Set(BlockHeaders.code)))
-    peer1.reply(PeerActor.MessageReceived(BlockHeaders(Seq(newBlockHeader))))
+    peer1TestProbe.expectMsg(PeerActor.SendMessage(GetBlockBodies(Seq(newBlockHeader.hash))))
+    peerMessageBus.expectMsg(Subscribe(MessageClassifier(Set(BlockBodies.code), PeerSelector.WithId(peer1.id))))
+    peerMessageBus.reply(MessageFromPeer(BlockBodies(Seq(BlockBody(Nil, Nil))), peer1.id))
 
-    peer1.expectMsg(PeerActor.SendMessage(GetBlockBodies(Seq(newBlockHeader.hash))))
-    peer1.expectMsg(PeerActor.Subscribe(Set(BlockBodies.code)))
-    peer1.reply(PeerActor.MessageReceived(BlockBodies(Seq(BlockBody(Nil, Nil)))))
+    peerMessageBus.expectMsg(Unsubscribe(MessageClassifier(Set(BlockHeaders.code), PeerSelector.WithId(peer1.id))))
+    peerMessageBus.expectMsg(Subscribe(MessageClassifier(Set(BlockHeaders.code), PeerSelector.WithId(peer2.id))))
+    peerMessageBus.expectMsg(Unsubscribe(MessageClassifier(Set(BlockBodies.code), PeerSelector.WithId(peer1.id))))
```

Comment:
Maybe we should do `expectMsgAllOf` between this last messages sent to `peerMessageBus`? This test seems to be randomly failing: 
```
[info] - should only ask a peer once for BlockHeaders in case execution of a block were to fail *** FAILED *** (12 milliseconds)
[info]   java.lang.AssertionError: assertion failed: expected Subscribe(MessageClassifier(Set(22),WithId(PeerId(testProbe-157)))), found Unsubscribe(Some(MessageClassifier(Set(20),WithId(PeerId(testProbe-157)))))
[info]   at scala.Predef$.assert(Predef.scala:219)
[info]   at akka.testkit.TestKitBase.expectMsg_internal(TestKit.scala:388)
[info]   at akka.testkit.TestKitBase.expectMsg(TestKit.scala:364)
[info]   at akka.testkit.TestKitBase.expectMsg$(TestKit.scala:364)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:814)
[info]   at io.iohk.ethereum.blockchain.sync.SyncControllerSpec$$anon$9.<init>(SyncControllerSpec.scala:561)
[info]   at io.iohk.ethereum.blockchain.sync.SyncControllerSpec.$anonfun$new$9(SyncControllerSpec.scala:522)
```